src/frontend/src/scheduler/worker_node_manager.rs

```diff
@@ -24,7 +26,14 @@ use crate::scheduler::{SchedulerError, SchedulerResult};
 
 /// `WorkerNodeManager` manages live worker nodes.
 pub struct WorkerNodeManager {
-    worker_nodes: RwLock<Vec<WorkerNode>>,
+    inner: RwLock<WorkerNodeManagerInner>,
+}
+
+#[derive(Default)]
+struct WorkerNodeManagerInner {
+    worker_nodes: Vec<WorkerNode>,
+    /// table vnode mapping info.
+    table_vnode_mapping: HashMap<TableId, Vec<ParallelUnitId>>,
```

Comment:
It seems better to maintain the mapping from **fragment_id** to `Vec<ParallelUnitId>`, but we may do this refactoring step by step.

Reply:
Another question: `table_vnode_mapping` is not persisted, right?

Reply:
How about assign a type alias to `Vec<ParallelUnitId>`?

Reply:
`table_vnode_mapping` is not persisted and built from notifications. Yes, the mapping is actually based on mapping from `fragment_id`, if so we need to store `fragment_id` in each `Table`.